lib/matplotlib/legend.py

```diff
@@ -517,8 +518,16 @@ def val_or_rc(val, rc_name):
             if not self.isaxes and loc in [0, 'best']:
                 loc = 'upper right'
 
+        type_err_message = ("loc must be string, coordinate tuple, or"
+                            f" an integer 0-10, not {loc!r}")
+
         # handle outside legends:
         self._outside_loc = None
+
+        # coerce list into tuple for backward compatibility
+        if isinstance(loc, list):
```

Comment:
```suggestion
        if isinstance(loc, collections.Iterable):
```
This would make it more general.  A list is the most obvious type of iterable someone might pass, but there may be others.  Your decision on the documentation makes sense to me.

Reply:
We seem to rely on `np.iterable` for this type of checks.

(It should probably be `collections.abc.Iterable` as well as importing directly is removed in 3.11.)

Reply:
Oh wait, we don’t want to convert all iterables here, because that would include strings and we clearly don’t want them to become tuples!  So I think this should come after the string handling…